src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +50,82 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(string value)
+        {
+            if (!this.AllowedValues.Contains(value))
+                return false;
 
-        [JsonProperty("allowedValues")]
-        public List<bool> AllowedValues { get; set; }
+            if ((value.Length < this.MinimumLength) || (value.Length > this.MaximumLength))
+                return false;
+
+            return true;
+        }
+
+        public override bool IsValid(bool value)
+        {
+            return false;
+        }
+
+        public override bool IsValid(int value)
+        {
+            return false;
+        }
     }
 
-    public class IntParameter : Parameter
+    public class BoolParameter : Parameter<bool>
     {
-        [JsonProperty("defaultValue")]
-        public int DefaultValue { get; set; }
+        public override bool IsValid(bool value)
+        {
+            return true;
+        }
 
-        [JsonProperty("allowedValues")]
-        public List<int> AllowedValues { get; set; }
+        public override bool IsValid(string value)
+        {
+            return false;
```

Comment:
Same discussion here: should we try to parse "true" or "false" first?

Reply:
What do you expect as truthy value vs. falsey value?

Reply:
If we decide not to try and parse for integers, we should probably do the same to remain consistent.

That being said, one thing I would definitely _not_ want to do is [something like the YAML spec](https://yaml.org/type/bool.html).

Reply:
I think the Go implementation for this is the most straightforward - https://github.com/deislabs/duffle/blob/master/pkg/bundle/parameters.go#L157-L163

Reply:
How about just `bool.TryParse(value)` then? This will return False for `"1"` though, if that's outside the scope, should be fine.